src/main/java/org/truffleruby/language/control/RaiseException.java

```diff
@@ -52,4 +54,42 @@ public Node getLocation() {
         return location;
     }
 
+    @Override
+    public Object getExceptionObject() {
+        return exception;
+    }
+
+    @Override
+    public boolean isSyntaxError() {
+        final RubyContext context = RubyLanguage.getCurrentContext();
+        return isA(context, context.getCoreLibrary().getSyntaxErrorClass());
+    }
+
+    @Override
+    public boolean isInternalError() {
+        final RubyContext context = RubyLanguage.getCurrentContext();
+        return isA(context, context.getCoreLibrary().getRubyTruffleErrorClass());
+    }
+
+    @Override
+    public boolean isExit() {
+        final RubyContext context = RubyLanguage.getCurrentContext();
+        return isA(context, context.getCoreLibrary().getSystemExitClass());
+    }
+
+    @Override
+    public int getExitStatus() {
+        final RubyContext context = RubyLanguage.getCurrentContext();
+        final Object status = context.send(exception, "status", null);
+        if (status instanceof Integer) {
+            return (int) status;
+        } else {
+            return 1;
```

Comment:
I think it's better to fail hard here (e.g. `throw new UnsupportedOperationException("SystemExit#status is not an Integer")` or just cast it unconditionally).